hw/ip/flash_ctrl/data/flash_ctrl.hjson

```diff
@@ -60,25 +60,26 @@
   ],
 
   param_list: [
```

Comment:
I would recommend the following two changes here:
1. Also add a local parameter `RegPagesPerBank` here, even though it is not used inside the register file. Right now, both these parameters are actually driven by topgen. But `NumBanks` propagates through reggen into the reg package and then into the main package. In contrast `PagesPerBank` is changed directly in the main package. I would treat them the same to simplify the templating and the documentation.

2. Separate the actually configurable parameters (`RegNumBanks`, `RegPagesPerBank`) below with comments saying that only these two are configurable and that they are defined by topgen. Also, I would put a comment saying that the others are not configurable, or if the user changes them, undefined behavior might result.

Reply:
this is a good suggestion, let me do that.